test/integration/metrics_service_integration_test.cc

```diff
@@ -0,0 +1,97 @@
+#include "common/common/version.h"
+#include "common/grpc/codec.h"
+#include "common/grpc/common.h"
+
+#include "test/integration/http_integration.h"
+
+#include "api/metrics_service.pb.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Stats {
+namespace Metrics {
+
+class MetricsServiceIntegrationTest : public HttpIntegrationTest,
+                                      public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  MetricsServiceIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam()) {}
+
+  void createUpstreams() override {
+    HttpIntegrationTest::createUpstreams();
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP2, version_));
+  }
+
+  void initialize() override {
+    config_helper_.addConfigModifier([](envoy::api::v2::Bootstrap& bootstrap) {
+      auto* metrics_service_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      metrics_service_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+      metrics_service_cluster->set_name("metrics_service");
+      metrics_service_cluster->mutable_http2_protocol_options();
+
+      auto* metrics_sink = bootstrap.add_stats_sinks();
+      // metrics_sink->MergeFrom(bootstrap.stat_sinks()[0]);
+      metrics_sink->set_name("envoy.metrics_service");
+      envoy::api::v2::MetricsServiceConfig config;
+      config.set_cluster_name("metrics_service");
+      MessageUtil::jsonConvert(config, *metrics_sink->mutable_config());
+
+    });
+
+    HttpIntegrationTest::initialize();
+    fake_upstreams_[1]->set_allow_unexpected_disconnects(true);
+  }
+
+  void waitForMetricsServiceConnection() {
+    fake_metrics_service_connection_ = fake_upstreams_[1]->waitForHttpConnection(*dispatcher_);
+  }
+
+  void waitForMetricsStream() {
+    metrics_service_request_ = fake_metrics_service_connection_->waitForNewStream(*dispatcher_);
+  }
+
+  void waitForMetricsRequest() {
```

Comment:
Can we potentially verify some actual metrics here?

Reply:
i can not verify actual metric names - but i can assert if there are metrics. I think that should be fine. LMK if you think otherwise.

Reply:
I think you should be able to verify at least a single gauge and counter exists, and maybe even that a failure is greater than zero. Can you dump them and find some good ones?

Reply:
i think we can actually validate the `metrics_service` cluster related metrics as we are adding the cluster in the test test it self. They will be there for sure. So i have added a validation for a `counter` and `gauge` including values. LMK if that makes sense. Sorry could have done this earlier